### ISSUE TITLE
IndexError when a world is generated with equal_exogenous_sales=True

### ISSUE THREAD

**Issue by a4rcvv** [yasserfarouk/scml#119]
**Describe the bug**

IndexError is raised when I generate `SCML2023OneShotWorld` with `equal_exogenous_sales=True` or `equal_exogenous_supply=True`.

**To Reproduce**

Generate a world like below.

```python
world = SCML2023OneShotWorld(
    **SCML2023OneShotWorld.generate(
        agent_types=seller_types + buyer_types,
        agent_processes=agent_processes,
        n_processes=2,
        n_steps=n_steps,
        random_agent_types=False,
        production_costs=2,
        exogenous_price_dev=0.0,
        equal_exogenous_sales=True,
        equal_exogenous_supply=True
    )
)
```

**Expected behavior**

A world is generated successfully.

**Actual behavior**

`IndexError` is raised.

```
  File "~/scml-a4rcvv/entry.py", line 100, in run
    **SCML2023OneShotWorld.generate(
  File "~/scml-a4rcvv/.venv/lib/python3.10/site-packages/scml/oneshot/world.py", line 914, in generate
    exogenous_supplies = distribute_quantities(
  File "~/scml-a4rcvv/.venv/lib/python3.10/site-packages/scml/common.py", line 186, in distribute_quantities
    return [np.asarray([values[_]] * a, dtype=int) for _ in range(n_steps)]
  File "~/scml-a4rcvv/.venv/lib/python3.10/site-packages/scml/common.py", line 186, in <listcomp>
    return [np.asarray([values[_]] * a, dtype=int) for _ in range(n_steps)]
IndexError: list index out of range
```

**Desktop (please complete the following information):**
 - OS: macOS 13.2.1 (Apple Silicon)
 - Python version: 3.10.10
 - scml version: 0.5.6

**Additional Context**

Possible workaround is to give `cap_exogenous_quantities=False` to `**SCML2023OneShotWorld.generate()`.

```python
world = SCML2023OneShotWorld(
    **SCML2023OneShotWorld.generate(
        agent_types=seller_types + buyer_types,
        agent_processes=agent_processes,
        n_processes=2,
        n_steps=n_steps,
        random_agent_types=False,
        production_costs=2,
        equal_exogenous_supply=True,
        equal_exogenous_sales=True,
        cap_exogenous_quantities=False
    )
)
```

It seems that, in [common.py](https://github.com/yasserfarouk/scml/blob/master/src/scml/common.py#L174-L186), `len(limit)` is equal to the number of *agents*, and `len(values)` is *originally* equal to `n_steps`. `zip(values, limit)` changes `len(values)` to the number of agents.

**Comment by yasserfarouk** [comment]
Fixed by 03bf21e

**Comment by yasserfarouk** [comment]
@a4rcvv diagnosed the issue correctly. We fixed it by removing the upper bound on quantities per agent from consideration when equal quantities is specified. 

This bug should not affect SCML 2023 though as in the competition we never use `equal=True` .
